editor/plugins/gdextension/gdextension_edit_dialog.cpp

```diff
@@ -0,0 +1,244 @@
+	vbox->add_child(scroll);
+
+	GridContainer *grid = memnew(GridContainer);
+	grid->set_columns(2);
+	grid->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	grid->set_v_size_flags(Control::SIZE_SHRINK_BEGIN);
+	scroll->add_child(grid);
+
+	// GDExtension file path (read-only).
+	Label *gdextension_path_label = memnew(Label);
+	gdextension_path_label->set_text(TTR("Path:"));
+	gdextension_path_label->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_RIGHT);
+	grid->add_child(gdextension_path_label);
+
+	gdextension_path = memnew(Label);
+	gdextension_path->set_text("res://addons/my_extension/my_extension.gdextension");
+	gdextension_path->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	grid->add_child(gdextension_path);
+
+	// Entry symbol.
+	Label *entry_symbol_label = memnew(Label);
+	entry_symbol_label->set_text(TTR("Entry Symbol:"));
+	entry_symbol_label->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_RIGHT);
+	grid->add_child(entry_symbol_label);
+
+	entry_symbol_edit = memnew(LineEdit);
+	entry_symbol_edit->set_placeholder("libname_library_init");
+	entry_symbol_edit->set_tooltip_text(TTR("Required. The symbol to use as the entry point for the extension."));
+	entry_symbol_edit->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	grid->add_child(entry_symbol_edit);
+
+	// Compatibility Min Version
+	Label *compat_min_version_label = memnew(Label);
+	compat_min_version_label->set_text(TTR("Compat Min Version:"));
+	compat_min_version_label->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_RIGHT);
+	grid->add_child(compat_min_version_label);
+
+	compat_min_version_edit = memnew(LineEdit);
+	compat_min_version_edit->set_placeholder(VERSION_BRANCH);
+	compat_min_version_edit->set_tooltip_text(TTR("Required. The minimum version of Godot that the extension is compatible with."));
+	compat_min_version_edit->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	grid->add_child(compat_min_version_edit);
+
+	// Compatibility Max Version
+	Label *compat_max_version_label = memnew(Label);
+	compat_max_version_label->set_text(TTR("Compat Max Version:"));
+	compat_max_version_label->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_RIGHT);
+	grid->add_child(compat_max_version_label);
+
+	compat_max_version_edit = memnew(LineEdit);
+	compat_max_version_edit->set_placeholder("<no maximum>");
+	compat_max_version_edit->set_tooltip_text(TTR("Optional. The maximum version of Godot that the extension is compatible with."));
+	compat_max_version_edit->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	grid->add_child(compat_max_version_edit);
+
+	// Reloadable checkbox.
+	Label *reloadable_label = memnew(Label);
+	reloadable_label->set_text(TTR("Reloadable:"));
+	reloadable_label->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_RIGHT);
+	grid->add_child(reloadable_label);
+
+	reloadable_checkbox = memnew(CheckBox);
+	reloadable_checkbox->set_tooltip_text(TTR("Optional. Whether the extension can be reloaded at runtime. Recommended to be true for most extensions."));
```

Comment:
I think it makes sense to just always use `true` for the `reloadable` flag (and not give users the option to set it). The initial GDExtension is _always_ reloadable at runtime. If the user makes significant enough changes that it isn't anymore, they can be expected to edit the `.gdextension` manually then to adjust the value.

Reply:
I disagree, this should be able to be customized in the UI. I don't want to force users to a text editor to change common settings

Reply:
What are some reasons to set `reloadable` to `false`? I thought we'd encourage making reloadable extensions whenever possible now.

Reply:
There are probably reasons, it's custom code after all. Who knows what people cook up in their extensions 😅
My own critique doesn't hold since this is the edit GUI, not the creation UI.